crates/vizia_core/src/views/switch.rs

```diff
@@ -0,0 +1,205 @@
+use crate::prelude::*;
+
+/// A Switch used to display and toggle boolean state.
+///
+/// Clicking on the Switch with the left mouse button triggers the `on_toggle` callback.
+///
+/// # Examples
+///
+/// ## Basic Switch
+///
+/// The Switch must bound to some boolean data.
+///
+/// ```
+/// # use vizia_core::prelude::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: bool,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: false }.build(cx);
+/// #
+/// Switch::new(cx, AppData::value);
+/// ```
+///
+/// ## Switch with an action
+///
+/// A Switch can be used to trigger a callback when toggled. Usually this is emitting an
+/// event responsible for changing the data the Switch is bound to.
+///
+/// ```
+/// # use vizia_core::prelude::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: bool,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # enum AppEvent {
+/// #     ToggleValue,
+/// # }
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: false }.build(cx);
+/// #
+/// Switch::new(cx, AppData::value).on_toggle(|cx| cx.emit(AppEvent::ToggleValue));
+/// ```
+///
+/// ## Switch with a label
+///
+/// A Switch is usually used with a label next to it describing what data the Switch
+/// is bound to or what the Switch does when pressed. This can be done, for example, by
+/// wrapping the Switch in an [`HStack`](crate::prelude::HStack) and adding a [`Label`](crate::prelude::Label)
+/// to it.
+///
+/// ```
+/// # use vizia_core::prelude::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: bool,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: false }.build(cx);
+/// #
+/// HStack::new(cx, |cx| {
+///     Switch::new(cx, AppData::value);
+///     Label::new(cx, "Press me");
+/// });
+/// ```
+///
+/// ## Custom Switch
+///
+/// The `text` modifier combined with a `map` on the lens can be used to customize the icon used by the Switch.
+///
+/// ```
+/// # use vizia_core::prelude::*;
+/// #
+/// # #[derive(Lens)]
+/// # struct AppData {
+/// #     value: bool,
+/// # }
+/// #
+/// # impl Model for AppData {}
+/// #
+/// # enum AppEvent {
+/// #     ToggleValue,
+/// # }
+/// #
+/// # let cx = &mut Context::new();
+/// #
+/// # AppData { value: false }.build(cx);
+/// # use vizia_core::fonts::unicode_names::CANCEL;
+///
+/// Switch::new(cx, AppData::value)
+///     .on_toggle(|cx| cx.emit(AppEvent::ToggleValue))
+///     .text(AppData::value.map(|flag| if *flag {CANCEL} else {""}));
+/// ```
```

Comment:
This doesn't apply to the switch and can be removed.

Reply:
Okay, I'll remove all coments as those can be written in the future with no issues

Reply:
Oh, it's only the highlighted section that needs to be removed (because the switch doesn't use text to display the inner part). The rest of the docs are fine.

Reply:
Okay. Changed!